### PR TITLE
Use Netty DefaultHttpHeaders by default for performance

### PR DESCRIPTION
**Pull request by testn**
Use DefaultHttpHeaders so that headers in HttpHeaders can be used directly. This helps Netty to serialize header name without having to encode. In addition to that, it can also use AsciiString.hashCode() for hash code for faster hash table lookup.

### CODE REVIEW

**Comment by vietj** [comment]
@testn have you measured a performance improvement ?

**Comment by testn** [comment]
It offers a few % improvement in request rate by reducing the CPU and memory consumption

**Comment by testn** [comment]
I updated it to contain the similar change to HTTP/2 as well

File: src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java [hunk 1]
```diff
@@ -113,29 +98,27 @@ public boolean isEmpty() {
...
+          }
+          @Override
+          public String next() {
+            return it.next().toString();
+          }
+        };
+      }
+      @Override
+      public int size() {
+        return nameSet.size();
```
Comment (testn) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: need to change this to ensure that the size() returns the correct value as you can see from testEntries() test

Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: good catch, I thought there was a copy like the previous issue

File: src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java [hunk 2]
```diff
@@ -35,26 +36,10 @@
...
-        }
-        buffer.setCharAt(index, (char)(c + ('a' - 'A')));
-      }
-    }
-    if (buffer != null) {
-      return buffer.toString();
-    } else {
-      return s;
-    }
+    return AsciiString.of(s).toLowerCase();
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: this creates an extra AsciiString object.

File: src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: add a test

File: src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java (hunk 2 above)
Comment (testn) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: fixed

File: src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java (hunk 1 above)
Comment (testn) [inline src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java]: the test is already on "testMultipleValues"

**Pull request closed without merging**